lib/pharos/phases/configure_dns.rb

```diff
@@ -6,7 +6,8 @@ class ConfigureDNS < Pharos::Phase
       title "Configure DNS"
 
       def call
-        patch_kubedns(
+        patch_deployment(
+          'coredns',
```

Comment:
Does this kube-dns -> coredns change cause downtime when upgrading from Pharos 1.2?

Reply:
Maybe, I didn't test to see how it behaves exactly.

If coredns is the new default, then I don't think we'd want to stick with kube-dns long-term. Options are:

* stick with just kube-dns for 1.2 and deal with it later
* mandatory migration to coredns for 1.2 
  * avoids any need to implement support for switching between kube-dns/coredns
* provide a gradual migration path for kube-dns -> coredns:
  * use coredns for new installs
  * keep kube-dns for existing installs
  * provide a `cluster.yml` knob to switch from kube-dns to coredns (or revert to `kube-dns` if coredns has some regressions)
  * eventually deprecate and remove `kube-dns`

Reply:
> keep kube-dns for existing installs

This would be non-trivial to implement, because it would require kube API access before `UpgradeMaster` / `ConfigureMaster` runs and generates the kubeadm config.

Reply:
> Does this kube-dns -> coredns change cause downtime when upgrading from Pharos 1.2?

Nope, it should be smooth ride (kube-dns deployment is removed after coredns is running).